GsaGH/Components/2_Geometry/Create2dElement.cs

```diff
@@ -46,7 +50,30 @@ protected override void SolveInstance(IGH_DataAccess da) {
         isLoadPanel = apiProperty2d.Type == Property2D_Type.LOAD;
       }
 
-      var elem = new GsaElement2d(ghmesh.Value, isLoadPanel);
+      GsaElement2d elem = null;
+      switch (geometry_parameter) {
+        case GH_Mesh mesh:
+          elem = new GsaElement2d(mesh.Value, isLoadPanel);
+          break;
+        case GH_Curve curve:
+          if (!isLoadPanel) {
+            throw new ArgumentException("Specify mesh geometry as the input parameter to create a 2D element.");
+          }
+          if (curve.Value.TryGetPolyline(out Rhino.Geometry.Polyline polyline)) {
+            if (polyline.ToArray().Length < 3) {
```

Comment:
Add a GH Warning

DA.SetWarning...

To mention that an attempt was made to automatically convert Curve to Polyline.

That being said, can you check what happens if the user provides a polyline? I think we need to make an attempt to first cast the Curve to the various other types: Polyline, PolyCurve and NurbsCurve, and handle each case separately. Check what currently happens if user provides a weird curve.

Reply:
if it becomes too complex, perhaps we can offload the logic for extracting the polyline to the user and ask for a true polyline to be provided.

Reply:
@psarras Lets discuss this once you are back